org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/wizards/AbapGitWizardSelectObjectsAndPull.java

```diff
@@ -0,0 +1,255 @@
+package org.abapgit.adt.ui.internal.wizards;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.abapgit.adt.backend.IApackManifest;
+import org.abapgit.adt.backend.IApackManifest.IApackDependency;
+import org.abapgit.adt.backend.IRepositoryService;
+import org.abapgit.adt.backend.RepositoryServiceFactory;
+import org.abapgit.adt.backend.model.abapgitrepositories.IRepositories;
+import org.abapgit.adt.backend.model.abapgitrepositories.IRepository;
+import org.abapgit.adt.backend.model.agitpullmodifiedobjects.IAbapGitObject;
+import org.abapgit.adt.backend.model.agitpullmodifiedobjects.IAbapGitPullModifiedObjects;
+import org.abapgit.adt.backend.model.agitpullmodifiedobjects.IAgitpullmodifiedobjectsFactory;
+import org.abapgit.adt.backend.model.agitpullmodifiedobjects.IOverwriteObjects;
+import org.abapgit.adt.backend.model.agitpullmodifiedobjects.IPackageWarningObjects;
+import org.abapgit.adt.ui.AbapGitUIPlugin;
+import org.abapgit.adt.ui.internal.i18n.Messages;
+import org.abapgit.adt.ui.internal.repositories.ModifiedObjectsForRepository;
+import org.abapgit.adt.ui.internal.wizards.AbapGitWizard.CloneData;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.Assert;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.jface.dialogs.DialogPage;
+import org.eclipse.jface.dialogs.IMessageProvider;
+import org.eclipse.jface.operation.IRunnableWithProgress;
+import org.eclipse.jface.wizard.IWizardContainer;
+import org.eclipse.jface.wizard.IWizardPage;
+import org.eclipse.jface.wizard.Wizard;
+import org.eclipse.jface.wizard.WizardDialog;
+import org.eclipse.jface.wizard.WizardPage;
+import org.eclipse.ui.plugin.AbstractUIPlugin;
+
+import com.sap.adt.tools.core.project.AdtProjectServiceFactory;
+
+public class AbapGitWizardSelectObjectsAndPull extends Wizard {
+
+	private final String destination;
+	private final CloneData cloneData;
+	Map<String, IAbapGitPullModifiedObjects> repoToSelectedObjectsMap;
+	private AbapGitWizardPageObjectsSelectionForPull pageOverwriteObjectsSelection;
+	private AbapGitWizardPageObjectsSelectionForPull pagePackageWarningObjectsSelection;
+	String transportRequest;
+
+	public AbapGitWizardSelectObjectsAndPull(IProject project, CloneData cloneData,
+			String transportRequest) {
+		this.destination = AdtProjectServiceFactory.createProjectService().getDestinationId(project);
+		this.cloneData = cloneData;
+		this.repoToSelectedObjectsMap = new HashMap<String, IAbapGitPullModifiedObjects>();
+		this.transportRequest = transportRequest;
+		fetchModifiedObjects(); //fetch the modified objects for repository and its dependencies
```

Comment:
1. In case a repository does not have any dependencies, there won't be  `AbapGitWizardPageApack` page. Hence logic for fetching modified objects for main repo is in `AbapGitWizardPageBranchAndPackage` page.
  2. In case a repository has dependencies, there will be  `AbapGitWizardPageApack` page. Hence logic for fetching modified objects for dependencies in `AbapGitWizardPageApack` page. In this case we can move logic of fetching modified objects for main repo in the `AbapGitWizardPageApack` from `AbapGitWizardPageBranchAndPackage` . But it will have to be there for case 1.
 
These methods are refactored to a common method `fetchAndExtractModifiedObjectsToPull` in RepositoryUtil class .